packages/trpc/server/routers/viewer/organizations/createPhoneCall.handler.ts

```diff
@@ -0,0 +1,175 @@
+import { z } from "zod";
+
+import { checkRateLimitAndThrowError } from "@calcom/lib/checkRateLimitAndThrowError";
+import { handleErrorsJson } from "@calcom/lib/errors";
+import logger from "@calcom/lib/logger";
+import type { PrismaClient } from "@calcom/prisma";
+import type { AIPhoneSettingSchema } from "@calcom/prisma/zod-utils";
+import type { TrpcSessionUser } from "@calcom/trpc/server/trpc";
+
+type CreatePhoneCallProps = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+    prisma: PrismaClient;
+  };
+  input: z.infer<typeof AIPhoneSettingSchema>;
+};
+
+const createRetellLLMSchema = z
+  .object({
+    llm_id: z.string(),
+    llm_websocket_url: z.string(),
+  })
+  .passthrough();
+
+const getRetellLLMSchema = z
+  .object({
+    general_prompt: z.string(),
+    begin_message: z.string().nullable(),
+  })
+  .passthrough();
+
+const createPhoneSchema = z
+  .object({
+    call_id: z.string(),
+    agent_id: z.string(),
+  })
+  .passthrough();
+
+const getPhoneNumberSchema = z
+  .object({
+    agent_id: z.string(),
+  })
+  .passthrough();
+
+export const fetcher = async (endpoint: string, init?: RequestInit | undefined) => {
+  return fetch(`https://api.retellai.com${endpoint}`, {
+    method: "GET",
+    headers: {
+      Authorization: `Bearer ${process.env.RETELL_AI_KEY}`,
+      "Content-Type": "application/json",
+      ...init?.headers,
+    },
+    ...init,
+  }).then(handleErrorsJson);
+};
+
+const createPhoneCallHandler = async ({ input, ctx }: CreatePhoneCallProps) => {
+  await checkRateLimitAndThrowError({
+    rateLimitingType: "core",
+    identifier: `createPhoneCall:${ctx.user.id}`,
+  });
+
+  const { yourPhoneNumber, numberToCall, guestName, eventTypeId, beginMessage, generalPrompt } = input;
+
+  const aiPhoneCallConfig = await ctx.prisma.aIPhoneCallConfiguration.upsert({
+    where: {
+      eventTypeId,
+    },
+    update: {
+      beginMessage,
+      generalPrompt,
+      isCalAiPhoneCallEnabled: true,
+    },
+    create: {
+      eventTypeId,
+      beginMessage,
+      generalPrompt,
+      isCalAiPhoneCallEnabled: true,
+    },
+  });
+
+  let llmWebSocketUrlToBeUpdated = null;
+
+  if (!aiPhoneCallConfig.llmId) {
+    const createdRetellLLM = await fetcher("/create-retell-llm", {
+      method: "POST",
+      body: JSON.stringify({
+        general_prompt: generalPrompt,
+        begin_message: beginMessage,
+        general_tools: [
+          {
+            type: "end_call",
+            name: "end_call",
+            description: "Hang up the call, triggered only after appointment successfully scheduled.",
+          },
+          {
+            type: "check_availability_cal",
+            name: "check_availability",
+            // TODO: CAL API KEY Environment Var
+            // cal_api_key: ,
```

Comment:
What Cal API Key we use here ?

Reply:
ahhh right. this has to be the key of the person with the event-type that we are accessing